grpcutil: Return true from RequestDidNotStart on open circuit breaker error

mneverov opened the pull request:
grpcutil: Return true from RequestDidNotStart on open circuit breaker error

Previously, the gRPC error check returned false on open circuit breaker error
that led to the request retries. Returning true instead should prevent
needless retries.

Fixes: #55869

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/59309)
<!-- Reviewable:end -->

tbg commented:
Thanks Max! I'll take a look next week.

mneverov committed:
grpcutil: Return true from RequestDidNotStart on open circuit breaker error

Previously, the gRPC error check returned false on open circuit breaker error
that led to the request retries. Returning true instead, should prevent
needless retries.

Fixes: #55869

Release note: None

tbg reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>, thanks Max!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/59309#-:-MRtgVf6CkouRx_YhSmZ:b-8a3nus)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @darinpp and @lunevalex)



<!-- Sent from Reviewable.io -->

tbg commented:
Btw, I just reached out on the community slack. Just mentioning this in case you don't get notified.
